pkg/cloud/aws/metadata/provider.go

```diff
@@ -0,0 +1,175 @@
+package metadata
```

Comment:
If i understand it correctly this is ec2 specific metadata ?
If so i would add another package layer so its /pkg/cloud/aws/metadata/ec2 (might be that we add such a provider aswell for elastic kubernetes service